src/psi_environment/data/map.py

```diff
@@ -1,29 +1,52 @@
+from typing import Type
+
 import numpy as np
 
 from psi_environment.data.car import Car, DummyAgent
 from psi_environment.data.map_state import MapState, Road
 from psi_environment.data.car import Action
+from psi_environment.data.point import Point
+
+DUMMY_ID_START = 2
 
 
 class Map:
-    def __init__(self, random_seed: int, n_bots: int = 3):
+    def __init__(self, random_seed: int, n_bots: int = 3, agent_type: Type[Car] = None, n_points: int = 3):
+        self.n_points = n_points
         self._map_state = MapState(random_seed)
+        self._points = []
 
         self._cars: list[Car] = []
 
         self._random_seed = random_seed
 
         # TODO: breaks if number of cars is greater than number of roads
         roads = self._map_state.get_roads()
-        road_idxs = np.random.choice(len(roads), size=n_bots)
+        road_idxs = np.random.choice(len(roads), size=n_bots + 1)
         road_keys = [list(roads)[idx] for idx in road_idxs]
 
         for i, road_key in enumerate(road_keys):
-            car_idx = i + 1
+            if i == 0:
+                car_idx = 1
+                road_pos_idx = self._map_state.add_car(road_key, car_idx)
+                car = agent_type(road_key, road_pos_idx, car_idx)
+                self._cars.append(car)
+                continue
+            car_idx = DUMMY_ID_START + i
             road_pos_idx = self._map_state.add_car(road_key, car_idx)
             car = DummyAgent(road_key, road_pos_idx, self._random_seed, car_idx)
             self._cars.append(car)
 
+        road_idxs = np.random.choice(len(roads), size=n_bots)
+        road_keys = [list(roads)[idx] for idx in road_idxs]
+
+        for i, road_key in enumerate(road_keys):
+            point_idx = i + 1
+            road = roads[road_key]
+            pos_idx = np.random.randint(road.length)
+            point = Point(road_key, pos_idx, point_idx)
+            self._points.append(point)
```

Comment:
Wydaje mi się że mówiliśmy o tym, żeby punkt związany był z miejscem na mapie jako współrzędnymi x,y a nie z drogą i pozycją, wiem że komplikuje to trochę logikę sprawdzania zbierania punktów, ale to daje nam furtkę do dodania kładzenia punktów również na skrzyżowaniach